src/coprocessor/codec/data_type/mod.rs

```diff
@@ -23,3 +27,100 @@ pub use crate::coprocessor::codec::mysql::{Decimal, Duration, Json, Time as Date
 // Dynamic eval types.
 pub use self::scalar::ScalarValue;
 pub use self::vector::VectorValue;
+pub use self::vector_like::{VectorLikeValueRef, VectorLikeValueRefSpecialized};
+
+/// A trait of evaluating current concrete eval type into a MySQL logic value, represented by
+/// Rust's `bool` type.
+pub trait AsMySQLBool {
+    /// Evaluates into a MySQL logic value.
+    fn as_mysql_bool(&self) -> bool;
+}
+
+impl AsMySQLBool for Int {
+    #[inline]
+    fn as_mysql_bool(&self) -> bool {
+        *self != 0
+    }
+}
+
+impl AsMySQLBool for Real {
+    #[inline]
+    fn as_mysql_bool(&self) -> bool {
+        self.round() != 0f64
+    }
+}
+
+impl AsMySQLBool for Bytes {
+    #[inline]
+    fn as_mysql_bool(&self) -> bool {
+        // FIXME: No unwrap?? No without_context??
+        !self.is_empty()
+            && crate::coprocessor::codec::convert::bytes_to_int_without_context(self).unwrap() != 0
```

Comment:
Here we need context, and shouldn't use `unwrap`

Reply:
You are right. It should be warnings (and no panics or errors).
It is also broken due to: https://github.com/tikv/tikv/issues/4327
Maybe we can fix this after #4327 is resolved. I will remember :)